opencga-catalog/src/main/java/org/opencb/opencga/catalog/managers/JobManager.java

```diff
@@ -613,6 +614,93 @@ public OpenCGAResult<Job> submit(String studyStr, String toolId, Enums.Priority
+            }
+            ParamUtils.checkDateIsNotExpired(scheduledStartTime, JobDBAdaptor.QueryParams.SCHEDULED_START_TIME.key());
+
+            ObjectMap params = new ObjectMap(JobDBAdaptor.QueryParams.SCHEDULED_START_TIME.key(), scheduledStartTime);
+            QueryOptions queryOptions = new QueryOptions();
+
+            if (StringUtils.isNotEmpty(eventMessage)) {
+                // Add event message
+                List<Event> eventList = Collections.singletonList(new Event(Event.Type.INFO, "reschedule", eventMessage));
+                params.append(JobDBAdaptor.QueryParams.INTERNAL_EVENTS.key(), eventList);
+
+                Map<String, Object> actionMap = new HashMap<>();
+                actionMap.put(JobDBAdaptor.QueryParams.INTERNAL_EVENTS.key(), ParamUtils.BasicUpdateAction.ADD);
+                queryOptions.getMap(Constants.ACTIONS, actionMap);
+            }
+
+            Query query = new Query()
+                    .append(JobDBAdaptor.QueryParams.STUDY_UID.key(), study.getUid())
+                    .append(JobDBAdaptor.QueryParams.UID.key(), jobUids);
+            OpenCGAResult<Job> update = getJobDBAdaptor(organizationId).update(query, params, queryOptions);
+
+            auditManager.audit(organizationId, userId, Enums.Action.RESCHEDULE_JOB, Enums.Resource.JOB, "", "", "",
+                    "", auditParams, new AuditRecord.Status(AuditRecord.Status.Result.SUCCESS));
+
+            return update;
+        } catch (Exception e) {
+            auditManager.audit(organizationId, userId, Enums.Action.RESCHEDULE_JOB, Enums.Resource.JOB, "", "", "",
+                    "", auditParams, new AuditRecord.Status(AuditRecord.Status.Result.ERROR, e));
+            throw e;
+        }
+    }
+
+    public OpenCGAResult<ExecutionTime> getExecutionTimeByMonth(String organizationId, Query query, String token) throws CatalogException {
+        JwtPayload tokenPayload = catalogManager.getUserManager().validateToken(token);
+        ParamUtils.checkParameter(organizationId, "organizationId");
+        authorizationManager.checkIsAtLeastOrganizationOwnerOrAdmin(organizationId, tokenPayload.getUserId(organizationId));
+        return getJobDBAdaptor(organizationId).executionTimeByMonth(query);
+    }
+
+    private void checkExecutionLimitQuota(String organizationId) throws CatalogException {
+        if (exceedsExecutionLimitQuota(organizationId)) {
+            throw new CatalogException("The organization '" + organizationId + "' has reached the maximum quota of execution hours ("
+                    + configuration.getQuota().getMaxNumJobHours() + ") for the current month.");
+        }
+    }
+
+    public boolean exceedsExecutionLimitQuota(String organizationId) throws CatalogException {
+        // Get current year/month
+        String time = TimeUtils.getTime(TimeUtils.getDate(), TimeUtils.yyyyMM);
+        Query query = new Query(JobDBAdaptor.QueryParams.MODIFICATION_DATE.key(), time);
```

Comment:
Query by ModificationDate? What if someone modifies the description or "visited" boolean of a 2 months old job? Would then that be counted for the current month'squota?

Reply:
Might this filter be by "execution end" ?

Reply:
You're right and done :)